src/views/MyPR.vue

```diff
@@ -37,13 +40,62 @@ import axios from "axios";
 import { formatDistanceToNow } from "date-fns";
 import {getCollection} from "../composables/getCollection";
 import { projectAuth } from "../firebase/config";
+import {projectFirestore} from "@/firebase/config";
 import Nav from "@/components/Nav.vue";
 
 export default {
     name: "MyPR",
     components: {
         Nav,
     },
+
+    methods :
+    {
+        set_id (id){
+            return id;
+        },
+
+        delete_pr (id) {
+            this.formattedUserPRData.filter(async (doc) => {
+                
+                if (doc.id == id){
+                    const colRef = await projectFirestore.collection("dashboard-2023")
+                    const docRef = colRef.doc(doc.id)
+                    docRef.delete().then(() => {
+                        console.log(docRef + " deleted")
+                    }).catch((e) => {
+                        console.log(e.message)
+                    })
+                    var toast = "Your PR " + doc.message + " is deleted successfully."
+                    alert(toast)
+                    console.log("finished deleting")
+                    
+                    
+                
+            }})
+            
+        },
+
+        async update_score (uid, difficulty) {
+            
+            var current_score = 0
+            var new_score = 0
+            var current_prs = 0
```

Comment:
@Doraemon012 Dont calll the methods in the `MyPR.vue` we have a seperate file for defining all the methods, see how we have defined it there and create your function in `src > composables. > deletedocuments.js`

We always try to refractor our code, so to have easy navigation and debugging, Therefore its a good practice to breakdown thingsj in different files instead of including them in a single file only.